src/Transition.js

```diff
@@ -483,11 +470,7 @@ Transition.propTypes = {
    *
    * @type {number | { enter?: number, exit?: number, appear?: number }}
    */
-  timeout: (props, ...args) => {
-    let pt = timeoutsShape
-    if (!props.addEndListener) pt = pt.isRequired
-    return pt(props, ...args)
-  },
+  timeout: timeoutsShape,
```

Comment:
**question:** why the change here? if we don't have a shim to add an end listener, then we should require a timeout to ensure that the transition end callback fires

Reply:
the "exit" callbacks, rather